assets/app/scripts/extensions/javaLink.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsoleExtensions.extension:JavaLinkController
+ * @description
+ * # ProjectController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsoleExtensions', ['openshiftConsole'])
+	.factory("JavaLinkUtils", function() {
+		var self = {
+			join: function() {
+        var paths = arguments;
+		    var tmp = [];
+		    var length = paths.length - 1;
+		    _.forEach(paths, function (path, index) {
+		      if (!path || path === "" || !angular.isString(path)) {
+		        return;
+		      }
+		      if (index !== 0 && _.startsWith(path, '/')) {
+		        path = path.slice(1);
+		      }
+		      if (index !== length && _.endsWith(path, '/')) {
+		        path = path.slice(0, path.length - 1);
+		      }
+		      if (path.length > 0) {
+		        tmp.push(path);
+		      }
+		    });
+		    var rc = tmp.join('/');
+		    return rc;
+			},
+			findParentWith: function($scope, attribute) {
+		    if (attribute in $scope) {
+		      return $scope;
+		    }
+		    if (!$scope.$parent) {
+		      return null;
+		    }
+		    // let's go up the scope tree
+		    return self.findParentWith($scope.$parent, attribute);
+			}
+		};
+		return self;
+	})
+  .factory("ProxyPod", function(API_CFG, JavaLinkUtils) {
+    return function(namespace, podId, port) {
+      var kubeProxyURI = new URI(window.location.protocol + '//' + API_CFG.k8s.hostPort + API_CFG.k8s.prefix);
+      var apiVersion = API_CFG.k8s.version || 'v1beta3';
+      if (port) {
+        podId = podId + ':' + port;
+      }
+      kubeProxyURI.path(JavaLinkUtils.join(kubeProxyURI.path(), apiVersion, 'proxy/namespaces', namespace, 'pods', podId));
+      return kubeProxyURI.toString();
+    };
+  })
+  .run(function(ProxyPod, BaseHref, JavaLinkUtils, HawtioExtension, $templateCache, $compile) {
+    var template =
+'<span ng-show="jolokiaUrl" title="View java details">' +
+'  <a ng-click="gotoContainerView($event, container, jolokiaUrl)" ng-href="jolokiaUrl">' +
+'    <i class="fa fa-external-link"></i>' +
+'  </a>' +
+'</span>';
+    HawtioExtension.add('container-links', function ($scope) {
+			var container = $scope.container;
+			if (!container) {
+				return;
+			}
+      var jolokiaPort = _.find((container.ports || []), function(port) {
+        return port.name && port.name.toLowerCase() === 'jolokia';
+      });
+			if (!jolokiaPort) {
+				return;
+			}
+      var pod = (JavaLinkUtils.findParentWith($scope, 'pod') || {}).pod;
```

Comment:
I'm wondering if `$scope.eval("pod")` would do the same thing... can you check? The angular docs made it sound like angular already handles inheritance to parent scopes when evaluating expressions...